### DISCUSSION TITLE
An error occurred during the code running for the latest Mtex (V2.3.0)

### DISCUSSION THREAD

**Discussion by FeC3-pearlite** [ORTools4MTEX/ORTools#78] [Q&A] [answered]
Dear developers,
Thank you very much for providing the powerful tool "ORTools" for EBSD analysis. There is a problem when I run the code of  "ORTools_Example10.m".
**Description of the bug**
When I run " [newGrains,~] = computeVariantGrains(job);" , an error occurred .
The error is "Error using cat
The dimensions of the arrays to be concatenated do not match.

Error in dynProp/cat (line 35)
dp.prop.(fn{i}) = cat(dim,s.(fn{i}));

Error in EBSD/cat (line 16)
ebsd = cat@dynProp(1,varargin{:});

Error in dynProp/vertcat (line 48)
dp = cat(1,varargin{:});

Error in computeVariantGrains (line 67)
newEBSD = [cEBSD; remainingEBSD];

Error in ORTools_example10 (line 125)
[newGrains,~] = computeVariantGrains(job);"

**ebsd data**
I have tried different data but the error is still existing. 

**MTEX version:**
 - Version [2.3.0]

I am looking forward to your reply.

**Comment by AzdiarGazder** [comment]
Hi @FeC3-pearlite,

Thank you for getting in touch and letting us know of a possible bug.

To reproduce your error and diagnose the fault, I'd be very grateful if you could please upload your script and the EBSD map being called for parent grain reconstruction by your script. To enable this, please post a weblink to download these files from.

Warm regards,
Azdi

**Comment by FeC3-pearlite** [comment]
Dear developers,
It's a surprise to receive the reply from you so quickly, thank you! 
There is a mistake in my statement last time. The Mtex version I have used is V5.10.0. 
I just used the original script named 'ORTools - Example 10' in the latest version of ORTools. I have upload the ebsd map and the script in the attachments. There is similarly a mistake in the function 'computeVariantGrains'. I also tried V2.2.0 ORTools based on Mtex 5.10.0, but the error doesn't occur. 
**EBSD map**
![parent2](https://github.com/user-attachments/assets/a911d988-a269-4d72-94cb-4b9dfe5641b7)
![parent](https://github.com/user-attachments/assets/fa9426d3-1810-402b-97b1-29dc8dd7d3b2)
![child ipf map](https://github.com/user-attachments/assets/b64ba1ca-ad4d-4807-90bc-8e7c1d33ce59)
**Script:**
https://github.com/FeC3-pearlite/ORTools-example-10/blob/main/ORTools_example10.m
I am looking forward to your reply.
Yours sincerely,
Jack Liu

**Comment by AzdiarGazder** [comment, marked as answer]
Hi @FeC3-pearlite,

I am debugging using:
- [MTEX 5.11.2](https://mtex-toolbox.github.io/download), and 
- the latest developer version of [ORTools](https://github.com/ORTools4MTEX/ORTools).

Based on the above, I could not replicate your error for [ORTools_example10]( https://github.com/ORTools4MTEX/ORTools/blob/develop/ORTools_example10.m).
I advise you to please download the above versions of MTEX and ORTools using the links I have provided.
Please let me know if the error turns up.

Hope this helps.

<ins>**Lastly, and importantly, if an explanation from the maintainers resolves your issue, please press the "Mark as answer" button on the appropriate response. Your help in this regard enables us to close discussions as complete. **</ins>

Warm regards,
Azdi

**Comment by FeC3-pearlite** [comment]
Dear developers,
Thanks for your advice, my problem has been solved. This error may be related to the version of Mtex. I tried Mtex 5.11.2 and launch the same result as you. 
Thank you very much!

Yours sincerely,
Jack Liu
